geode-core/src/main/java/org/apache/geode/cache/control/RestoreRedundancyResults.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.cache.control;
+
+import java.util.Map;
+
+import org.apache.geode.cache.partition.PartitionRebalanceInfo;
+import org.apache.geode.internal.cache.control.RestoreRedundancyRegionResult;
+import org.apache.geode.internal.cache.partitioned.PartitionedRegionRebalanceOp;
+
+/**
+ * A class to collect the results of restore redundancy operations for one or more regions and
+ * determine the success of failure of the operation.
+ */
+public interface RestoreRedundancyResults {
+
+  /**
+   * {@link #SUCCESS} is defined as every included region having at least one redundant copy.
```

Comment:
The idea is that as long as redundancy has been established, at any level, then the operation has succeeded, since in 99% of cases, a user cares far more about whether they have redundancy or not vs how much redundancy they have. A user can get information about how many regions (if any) do not have fully satisfied redundancy using the `RestoreRedundancyResults.getUnderRedundancyRegionResults()` method if they need that information.

Reply:
Sounds good, thanks for the clarification!.

Reply:
After a bit more discussion, it was decided to go with the suggestion here. `SUCCESS` will now only be returned if all regions have their redundancy fully satisfied.